src/homelogstore/log_store.cpp

```diff
@@ -237,13 +237,54 @@ void HomeLogStore::truncate(const logstore_seq_num_t upto_seq_num, const bool in
     if (locked_now) { m_logdev.unlock_flush(); }
 }
 
+void HomeLogStore::truncate_sync(const logstore_seq_num_t upto_seq_num, const bool in_memory_truncate_only) {
```

Comment:
I would rather do fast_forward api which does fill_gap and then call truncate this. Also we don't need in_memory_truncate_only for this use case.